core/src/main/scala/neotypes/DeferredQuery.scala

```diff
@@ -525,26 +271,29 @@ private[neotypes] object DeferredQuery {
 final class DeferredQueryBuilder private[neotypes] (private val parts: List[DeferredQueryBuilder.Part]) {
   import DeferredQueryBuilder.{PARAMETER_NAME_PREFIX, Param, Part, Query, SubQueryParam}
 
-  /** Creates a [DeferredQuery] from this builder.
-    *
-    * @tparam T the result type of the constructed query.
-    * @return A [DeferredQuery[T]]
-    */
-  def query[T]: DeferredQuery[T] = {
+  def command[T <: Any]: DeferredCommand[T] = {
+    basebuilder[T, DeferredCommand[T]]((query, params, paramLocations) => DeferredCommand(query, params, paramLocations))
+  }
+
+  def query[T <: Any]: DeferredQuery[T] = {
+    basebuilder[T, DeferredQuery[T]]((query, params, paramLocations) => DeferredQuery(query, params, paramLocations))
+  }
+
+  def basebuilder[T <: Any, D <: Deferred[T]](f: (String, Map[String, QueryParam], List[Int]) => D): D = {
```

Comment:
I would not complicate this that much with that `f` function, but rather just make this method return the triplet `(String, Map[String, QueryParam], List[Int])`